### PR TITLE
Update: Discord Modules Documentation and Tutorials Addition

### PR DESCRIPTION
**Pull request by ImDarkLaw**
## Changes

- Expanded Discord and Discord Link modules documentation with additional content.
- Introduced a new Tutorials page on the sidebar, featuring comprehensive tutorials for the Discord and Discord Link addons.

*Essentials Discord configuration section remains a WIP, till we figure out how want to tackle it.*

### CODE REVIEW

File: docs/modules/discord.mdx [hunk 1]
```diff
@@ -2,8 +2,365 @@
...
 This addon requires you to have a matching version of EssentialsX installed.
 :::
 
-## Frequently Asked Questions
\ No newline at end of file
+EssentialsX Discord offers a simple, lightweight, user-friendly, and seamless bridge between Discord and Minecraft.
+
+## Features
+
+This module offers *essentials* features you'd want from a Discord bridge, such as:
```
Comment (Evidentsinger14) [inline docs/modules/discord.mdx]: as cleaver as the wordplay is, i'm not a fan of the emphasis + "s" on "essential" (as seen on the current wiki)

File: docs/modules/discord.mdx [hunk 2]
```diff
@@ -2,8 +2,365 @@
...
+
+- Minecraft chat to Discord channel
+- Discord channel to Minecraft
+- Minecraft to Discord event monitoring (Join/Leave/Death/Mute)
+- Server console relay to Discord.
+- Discord slash commands.
+- …and more.
+
+:::tip
+Just installed EssentialsDiscord? Check out our comprehensive [tutorials page](../../tutorials/discordtut) for step-by-step instructions on getting the plugin up and running smoothly!
```
Comment (Evidentsinger14) [inline docs/modules/discord.mdx]:
```diff
- tutorial page
+ tutorial
```

File: docs/modules/discord.mdx [hunk 3]
```diff
@@ -2,8 +2,365 @@
...
+
+## Configuration
+You can access the EssentialsDiscord configuration by opening `/plugins/EssentialsDiscord/config.yml`.
+
+
+---
+
+## Console Relay
+The console relay is pretty self-explanatory: it relays everything on your console into a Discord channel of
+your choosing. The console relay is ridiculously easy to set up and if your server is already running, you don't
```
Comment (Evidentsinger14) [inline docs/modules/discord.mdx]:
```diff
- ridiculously 
+ incredibly
```

File: docs/modules/discord.mdx [hunk 4]
```diff
@@ -2,8 +2,365 @@
...
+need to reload it!
```
Comment (Evidentsinger14) [inline docs/modules/discord.mdx]:
```diff
- reload
+ restart
```
 ? I'm not sure why this says reload, i know it's from the current docs, but you have to run `/ess reload` for changes to take effect.

File: docs/modules/discord.mdx [hunk 5]
```diff
@@ -2,8 +2,365 @@
...
+settings. Otherwise, if you'd like to see what other options you can use to customize console output, stick around.
+
+4. The first thing you can customize is the format of the message sent to Discord. By default, the timestamp,
+level (info/warn/error/etc.), and message are shown for each console message. Let's say you wanted to make the
+timestamp and level bold: since this message would be using Discord's markdown, we can just add \*\* to both sides of
+level and timestamp. Then, once you've done that, just do `/ess reload` and you should see your changes on Discord.
+> ![Bold Format](../../static/img/discord/bold-format.gif)
+
+5. Next, you can also configure the name you wish the to show above console messages. By default, it's "EssX Console
+Relay" but can be switched to anything you want.
```
Comment (Evidentsinger14) [inline docs/modules/discord.mdx]:
```diff
- switched
+ changed
```

File: docs/modules/discord.mdx [hunk 6]
```diff
@@ -2,8 +2,365 @@
...
+restricted to the console channel.
+> ![Command Relay](../../static/img/discord/command-relay.gif)
+
+7. That's all the options for the command relay!
+
+---
+
+## Configuring Messages
+EssentialsX Discord aims to keep its message-type system basic enough that simple things take little changes, while
+giving more fine grain control to those you want it.
```
Comment (Evidentsinger14) [inline docs/modules/discord.mdx]:
```diff
- you
+ who
```

File: docs/modules/discord.mdx [hunk 7]
```diff
@@ -2,8 +2,365 @@
...
+
+To give you a general overview of the system, EssentialsX Discord allows you to define different channel IDs in the
+`channels` section of the config. By default, two channels are pre-populated in the `channels` section, `primary`
+and `staff`. If you only completed the initial setup, the `staff` channel definition is all zeros. This is fine in
+most situations however, as the message system will always fall back to the `primary` channel if a channel ID is
+invalid.
+
+Now on to the types of messages you can receive themselves (which is where you're going to use these channel
```
Comment (Evidentsinger14) [inline docs/modules/discord.mdx:86]: I'm gonna be honest. I don't really understand what this means
" messages you can receive themselves "

File: docs/modules/discordlink.mdx [hunk 8]
```diff
@@ -20,6 +20,10 @@ EssentialsX Discord Link offers features such as:
 - Prevent unlinked players from moving/chatting
 - …and more
 
+:::tip
+Just installed EssentialsDiscordLink? Check out our comprehensive [tutorials page](../../tutorials/discordlinktut) for step-by-step instructions on getting the plugin up and running smoothly!
```
Comment (Evidentsinger14) [inline docs/modules/discordlink.mdx]:
```diff
- tutorials page
+ tutorial
```

File: docs/tutorials/discordlinktut.mdx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+1. This assumes the server has started and you have joined the server.
+
+2. Once on the server, run `/link` in Minecraft and take note of the code if gives you.
+   > ![Run /link](https://i.imgur.com/1EdqdOa.gif)
+
+3. Next, all you have to do is run the `/link` command in discord with the code provided.
+   > ![Run /link in Discord](https://i.imgur.com/yXkvMDX.gif)
+
+4. That's it! Now that you've learned how to link an account, go back up to the
+[EssentialsDiscordLink documentation](../../modules/discordlink) to see what else you can do!
```
Comment (Evidentsinger14) [inline docs/tutorials/discordlinktut.mdx]:
```diff
- [EssentialsDiscordLink documentation]
+ [Discord Link Docs]
```

File: docs/tutorials/discordtut.mdx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+From this screen, you'll need to copy your "Application ID" and save it for a later step. To copy your Application ID, click the upper-left most blue "Copy" button. Make sure to save it for a later step.
+> ![Copy Application ID](../../static/img/discord/copy-client-id.gif)
+> `Copy` -> Paste into Notepad for later step
+
+3. Optionally, you can set an icon for your application as it will be the icon for the bot too.
+> ![Avatar](../../static/img/discord/avatar.png)
+
+[//]: # (Deprecated)
+
+[//]: # (> ![Create Bot]&#40;../../static/img/discord/create-bot.gif&#41;)
```
Comment (Evidentsinger14) [inline docs/tutorials/discordtut.mdx]: I'm not sure what these are here for? surely they can be removed?

**Review by Evidentsinger14 — CHANGES_REQUESTED** [review]

File: docs/modules/discord.mdx (hunk 7 above)
Comment (ImDarkLaw) [reply docs/modules/discord.mdx:86]: I'm not sure either. Leaving it as it is for now, as I don't have much time to revisit the "Configuring Messages" section. Maybe we can poke JRoy about it?

File: docs/tutorials/discordtut.mdx (hunk 10 above)
Comment (ImDarkLaw) [reply docs/tutorials/discordtut.mdx]: Resolved as of latest commit.

File: docs/tutorials/discordlinktut.mdx (hunk 9 above)
Comment (ImDarkLaw) [reply docs/tutorials/discordlinktut.mdx]: Resolved as of latest commit.

File: docs/modules/discordlink.mdx (hunk 8 above)
Comment (ImDarkLaw) [reply docs/modules/discordlink.mdx]: Resolved as of latest commit.

File: docs/modules/discord.mdx (hunk 6 above)
Comment (ImDarkLaw) [reply docs/modules/discord.mdx]: Resolved as of latest commit.

File: docs/modules/discord.mdx (hunk 5 above)
Comment (ImDarkLaw) [reply docs/modules/discord.mdx]: Resolved as of latest commit.

File: docs/modules/discord.mdx (hunk 4 above)
Comment (ImDarkLaw) [reply docs/modules/discord.mdx]: Resolved as of latest commit.

File: docs/modules/discord.mdx (hunk 3 above)
Comment (ImDarkLaw) [reply docs/modules/discord.mdx]: Resolved as of latest commit.

File: docs/modules/discord.mdx (hunk 2 above)
Comment (ImDarkLaw) [reply docs/modules/discord.mdx]: Rssolved as of latest commit.

File: docs/modules/discord.mdx (hunk 1 above)
Comment (ImDarkLaw) [reply docs/modules/discord.mdx]: Resolved as of latest commit. We may want to revisit and change later.

**Review by Evidentsinger14 — APPROVED** [review]

**Pull request merged**